src/instrumenter/transpile.ts

```diff
@@ -236,13 +238,30 @@ function transpileId(expr: SId, ctx: TranspilingContext): Expression {
     if (expr.defSite instanceof VariableDeclaration) {
         if (expr.name !== expr.defSite.name) {
             assert(
-                expr.defSite.stateVariable,
+                expr.defSite.stateVariable || expr.defSite.vScope instanceof SourceUnit,
                 `Internal error: variable id ${expr.pp()} has different name from underlying variable ${expr.defSite.print()}` +
-                    `Variable renaming only allowed for public state vars with maps`
+                    `Variable renaming only allowed for public state vars with maps and imported global vars`
             );
         }
 
-        const res = factory.makeIdentifierFor(expr.defSite);
+        let res: Identifier;
+        // If the scribble name doesn't match the name of the underlying definition there are several cases:
+        // 1) This is a state variable - pick the underlying definition, since
+        //  we sometimes rename public state vars during instrumentation to allow
+        //  for an explicit getter fun
+        // 2) Global symbol in flat/json mode - pick the underlying definition, since all definitions are
+        //  flattened and renamed to avoid collisions
+        // 3) Global symbol in files mode - pick the scribble identifier - this is a case of import renaming
+        // 4) All other cases - pick the Scribble name
+        if (
+            expr.name !== expr.defSite.name &&
+            (expr.defSite.stateVariable ||
+                (expr.defSite.vScope instanceof SourceUnit && ctx.instrCtx.outputMode !== "files"))
+        ) {
```

Comment:
@blitz-1306 this if is kinda complex and I am not 100% sure of the logic

Reply:
Yeah. This looks like very complex thing. Maybe we would consider some samples to figure out edge-cases? Maybe do this as a follow-up?

Reply:
Yeah lets deal with this later